lib/wicked/IPv6.pm

```diff
@@ -0,0 +1,133 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper module for IPv6 tests
+# Maintainer: Jose Lausuch <jalausuch@suse.com>
+
+package wicked::IPv6;
+
+use Mojo::Base 'wickedbase';
+use testapi;
+
+sub evaluate_step {
+    my ($self, $cmd, $value, $expected) = @_;
+    my $output = script_output($cmd);
+    record_info('output', "cmd: $cmd \noutput:\n$output");
+    die("$value is not expected") if ($expected == 0 && $output =~ $value);
+    die("$value is expected")     if ($expected == 1 && $output !~ $value);
+}
+
+sub setup_interface {
+    my ($self, %args) = @_;
+    die '$args{iface} is required'      unless $args{iface};
+    die '$args{ifcfg_type} is required' unless $args{ifcfg_type};
+    record_info("STEP", 'Setup ' . $args{ifcfg_type} . ' interface ' . $args{iface});
+    $self->get_from_data('wicked/' . $args{ifcfg_type} . '_address/ifcfg-eth0', '/etc/sysconfig/network/ifcfg-' . $args{iface});
+    $self->wicked_command('ifup', $args{iface});
+}
+
+sub wait_for_router_announcement {
+    my ($self, %args) = @_;
+    die '$args{iface} is required' unless $args{iface};
+    record_info("STEP", 'I let ' . $args{iface} . ' wait for a router announcement');
+    assert_script_run('rdisc6 ' . $args{iface});
+}
+
+sub retrigger_dhcp_requests {
+    my ($self, %args) = @_;
+    die '$args{iface} is required' unless $args{iface};
+    record_info("STEP", 'I retrigger DHCP requests');
+    $self->wicked_command('ifdown', $args{iface});
+    $self->wicked_command('ifup',   $args{iface});
+}
+
+sub obtain_address_by_dhcpv4 {
+    my ($self, %args) = @_;
+    die '$args{iface} is required' unless $args{iface};
+    my $expected = $args{should_not} == 1 ? 0 : 1;
+    record_info("STEP", 'I ' . ($expected ? 'should' : 'should not') . ' obtain an address by DHCPv4');
+    # ip address show dev eth0
+    # out.should include "inet #{DHCP4_SUT0}"
+    # out.should_not include "inet #{DHCP4_SUT0}"
+    my $cmd = 'ip address show dev ' . $args{iface};
+    $self->evaluate_step($cmd, $self->get_ip(type => 'dhcp'), $expected);
+}
+
+sub obtain_autonomous_ipv6_address {
+    my ($self, %args) = @_;
+    die '$args{iface} is required' unless $args{iface};
+    my $expected = $args{should_not} == 1 ? 0 : 1;
+    record_info("STEP", 'I ' . ($expected == 1 ? 'should' : 'should not') . ' obtain an an autonomous IPv6 address');
+    # ip -6 address show dev eth0
+    # out.should include "inet6 #{RADVD_SUT0}"
+    # out.should_not include "inet6 #{RADVD_SUT0}"
+    my $cmd = 'ip -6 address show dev ' . $args{iface};
```

Comment:
so this approach actually lead us for more execution time , old version was doing one `script_run` and than query result , here to be be generic we paying with `script_output` for **every** check

Reply:
After several tests, it's not very time consuming, but if you think it's an issue I can change the way we execute commands and do it in the test itself and send the output directly to the function checks in ipv6.pm

Reply:
no let's leave it like this , maybe just use short form of command ? 
```suggestion
    my $cmd = 'ip -6 a s dev ' . $args{iface};
```

Reply:
> no let's leave it like this , maybe just use short form of command ?

I would vote for long forms everywhere it is possible. It is straightforward and more convenient to read.